webapp/src/dogma/common/components/NewFile.tsx

```diff
@@ -0,0 +1,187 @@
+    formState: { errors },
+  } = useForm<FormData>();
+  const dispatch = useAppDispatch();
+  const onSubmit = async (formData: FormData) => {
+    const data = {
+      commitMessage: {
+        summary: formData.summary,
+        detail: formData.detail,
+        markup: markup,
+      },
+      changes: [
+        {
+          path: '/' + formData.name, // TODO: Allow the actual path in the input form i.e. allow slash /
+          type: formData.type,
+          content: editorRef.current.getValue(),
+        },
+      ],
+    };
+    try {
+      const response = await addNewFle({ projectName, repoName, data }).unwrap();
+      if ((response as { error: FetchBaseQueryError | SerializedError }).error) {
+        throw (response as { error: FetchBaseQueryError | SerializedError }).error;
+      }
+      Router.push(`/app/projects/${projectName}/repos/${repoName}/list/head/`);
+      reset();
+      dispatch(
+        createMessage({
+          title: 'New file created',
+          text: `Successfully created ${formData.name}`,
+          type: 'success',
+        }),
+      );
+    } catch (error) {
+      dispatch(
+        createMessage({
+          title: 'Failed to create a new file',
+          text: ErrorHandler.handle(error),
+          type: 'error',
+        }),
+      );
+    }
+  };
+  const [markup, setMarkup] = useState('PLAINTEXT');
+  const editorRef = useRef(null);
+  const handleEditorMount: OnMount = (editor) => {
+    editorRef.current = editor;
+  };
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Flex minWidth="max-content" alignItems="center" mb={4}>
+        <Heading size="lg">Create new file</Heading>
+      </Flex>
+      <Box p={4}>
+        <RadioGroup onChange={setMarkup} value={markup} mb={2}>
+          <Flex gap={2}>
+            <Spacer />
+            <Radio value="PLAINTEXT" colorScheme="teal">
+              plain text
+            </Radio>
+            <Radio value="MARKDOWN" colorScheme="teal">
+              markdown
+            </Radio>
+          </Flex>
+        </RadioGroup>
+        <VStack>
+          <FormControl isInvalid={errors.name ? true : false} isRequired>
+            <FormLabel>Path</FormLabel>
+            {/* TODO: Allow slash / in the file path/name, i.e.exclude this input from the document hotkey */}
+            <Input
+              type="text"
+              placeholder="my-file-name"
+              {...register('name', { pattern: FILE_PATH_PATTERN })}
+            />
+            {errors.name && <FormErrorMessage>Invalid file name</FormErrorMessage>}
+          </FormControl>
+          <FormControl>
+            <FormLabel>Type</FormLabel>
+            <Select {...register('type')} defaultValue="UPSERT_JSON" name="type">
+              <option key="UPSERT_JSON" value="UPSERT_JSON">
+                JSON
+              </option>
+              <option key="UPSERT_TEXT" value="UPSERT_TEXT">
+                TEXT
+              </option>
+            </Select>
```

Comment:
It seems redundant since we only allow `.json` extension for JSON.
What do you think of setting `type` automatically depending on the file extension?

Reply:
Updated.